source/acquaman/BioXAS/BioXASXASScanActionController.cpp

```diff
@@ -0,0 +1,380 @@
+
+	notes.append(QString("SR1 Current:\t%1 mA\n").arg(QString::number(CLSStorageRing::sr1()->ringCurrent(), 'f', 1)));
+
+	// Note the mono settling time, if applicable.
+
+	BioXASSSRLMonochromator *mono = BioXASBeamline::bioXAS()->mono();
+	if (mono) {
+		double settlingTime = mono->braggMotor()->settlingTime();
+		if (settlingTime > 0)
+			notes.append(QString("Settling time:\t%1 s\n").arg(settlingTime));
+	}
+
+	return notes;
+}
+
+AMAction3* BioXASXASScanActionController::createInitializationActions()
+{
+	AMSequentialListAction3 *result = 0;
+
+	// Initialize the scaler.
+
+	AMSequentialListAction3 *scalerInitialization = 0;
+	CLSSIS3820Scaler *scaler = CLSBeamline::clsBeamline()->scaler();
+
+	if (scaler) {
+		double regionTime = double(bioXASConfiguration_->scanAxisAt(0)->regionAt(0)->regionTime());
+
+		scalerInitialization = new AMSequentialListAction3(new AMSequentialListActionInfo3("BioXAS Scaler Initialization Actions", "BioXAS Scaler Initialization Actions"));
+		scalerInitialization->addSubAction(scaler->createContinuousEnableAction3(false));
+		scalerInitialization->addSubAction(scaler->createDwellTimeAction3(regionTime));
+		scalerInitialization->addSubAction(scaler->createStartAction3(true));
+		scalerInitialization->addSubAction(scaler->createWaitForDwellFinishedAction(regionTime + 5.0));
+	}
+
+	// Initialize Ge 32-el detector, if using.
+
+	AMSequentialListAction3 *geDetectorInitialization = 0;
+	BioXAS32ElementGeDetector *geDetector = BioXASBeamline::bioXAS()->ge32ElementDetector();
+
+	if (geDetector) {
+		bool usingGeDetector = (bioXASConfiguration_->detectorConfigurations().indexOf(geDetector->name()) != -1);
+
+		if (usingGeDetector) {
+			geDetectorInitialization = new AMSequentialListAction3(new AMSequentialListActionInfo3("BioXAS Xpress3 Initialization", "BioXAS Xpress3 Initialization"));
+			geDetectorInitialization->addSubAction(geDetector->createDisarmAction());
+			geDetectorInitialization->addSubAction(geDetector->createFramesPerAcquisitionAction(int(bioXASConfiguration_->scanAxisAt(0)->numberOfPoints()*1.1)));	// Adding 10% just because.
+			geDetectorInitialization->addSubAction(geDetector->createInitializationAction());
+		}
+	}
+
+	// Initialize the mono.
+
+	AMSequentialListAction3 *monoInitialization = 0;
+	BioXASMonochromator *mono = BioXASBeamline::bioXAS()->mono();
+
+	if (mono) {
+
+		// Set the bragg motor power to PowerOn, must be on to move/scan.
+		CLSMAXvMotor *braggMotor = qobject_cast<CLSMAXvMotor*>(mono->braggMotor());
+
+		if (braggMotor) {
+			monoInitialization = new AMSequentialListAction3(new AMSequentialListActionInfo3("BioXAS Monochromator Initialization", "BioXAS Monochromator Initialization"));
+			monoInitialization->addSubAction(braggMotor->createPowerAction(CLSMAXvMotor::PowerOn));
+		}
+	}
+
+	// Initialize the standards wheel.
+
+	AMAction3* standardsWheelInitialization = 0;
+//	CLSStandardsWheel *standardsWheel = BioXASBeamline::bioXAS()->standardsWheel();
```

Comment:
The standards wheel is disabled on the beamline temporarily. As far as I know, there is no problem with the device itself. A better thing to do might be to comment out the creation of the standards wheel in BioXASSideBeamline, instead of the initialization actions here.